static/flow/views/program-editor-panel.js

```diff
@@ -929,34 +927,46 @@ var ProgramEditorPanel = function(options) {
     //
     this.nameHash = {};
 
+    //
+    // Recompute nameHash object which stores sensor and timer block names
+    //
+    this.updateNameHash = function() {
+        _this.nameHash = {};
+        for (var i = 0; i < _this.m_diagram.blocks.length; i++) {
+            var block = _this.m_diagram.blocks[i];
+              if (_this.isDeviceBlock(block.type) || block.type === 'timer') {
+                  _this.nameHash[block.name] = block;
+              }
+        }
+    };
+
     //
     // Determine if a block represents a physical sensor device. List of device blocks defined in utils/definitions
     //
     this.isDeviceBlock = function (type) {
-      return DEVICE_BLOCKS.indexOf(type) > -1
+      return DEVICE_BLOCKS.indexOf(type) > -1;
     };
 
     //
     // Determine if a block represents a filter. List of filter blocks defined in utils/definitions
     //
     this.isFilterBlock = function (type) {
       return FILTER_BLOCKS.indexOf(type) > -1;
-    }
+    };
 
     //
     // Used to create unique names for blocks
     //
     this.getUniqueName = function(name) {
-
         var block = this.nameHash[name];
         if (!block) {
             return name;
         }
         var count = 2;
-        while(this.nameHash[name + count]) {
```

Comment:
@dstrawberrygirl this is the key to make multiple sensors work.  The name given to the block must match the name in the flow client.  The flow client uses spaces so we're matching that format here.